src/modules/fancyzones/dll/dllmain.cpp

```diff
@@ -77,6 +76,19 @@ class FancyZonesModule : public PowertoyModuleIface
         {
             Trace::FancyZones::EnableFancyZones(true);
             m_app = MakeFancyZones(reinterpret_cast<HINSTANCE>(&__ImageBase), m_settings);
+
+            s_llKeyboardHook = SetWindowsHookEx(WH_KEYBOARD_LL, LowLevelKeyboardProc, GetModuleHandle(NULL), NULL);
+            if (!s_llKeyboardHook)
+            {
+                MessageBoxW(NULL, L"Cannot install keyboard listener.", L"PowerToys - FancyZones", MB_OK | MB_ICONERROR);
+            }
+
+            s_winEventHook = SetWinEventHook(EVENT_MIN, EVENT_MAX, nullptr, WinHookProc, 0, 0, WINEVENT_OUTOFCONTEXT | WINEVENT_SKIPOWNPROCESS);

```

Comment:
In the [original proposal](https://github.com/microsoft/PowerToys/issues/1264), the idea was to not just move the hook registration to FZ module, but reduce the performance impact of it by doing 2 things:
- [subscribing only to a certain events](https://github.com/TheMrJukes/PowerToys/compare/master...TheMrJukes:user/bretan/pt-optimize#diff-d26a7704aedb11415d1478eb46662ac0R121), not the whole range of them. That would mean calling `SetWinEventHook` multiple times for each event type [mentioned in `get_winhook_events`](https://github.com/TheMrJukes/PowerToys/compare/master...TheMrJukes:user/bretan/pt-optimize#diff-d26a7704aedb11415d1478eb46662ac0R125-R130).
- [dynamically (un)subscribe to a certain events](https://github.com/TheMrJukes/PowerToys/compare/master...TheMrJukes:user/bretan/pt-optimize#diff-d26a7704aedb11415d1478eb46662ac0R284-R291). That would mean modifying `WinHookProc` callback to add (un)subscribtion to `EVENT_OBJECT_LOCATIONCHANGE` when receiving `EVENT_SYSTEM_MOVESIZESTART`/`EVENT_SYSTEM_MOVESIZEEND` events.

I wonder if we should do it as part of this PR?

Reply:
I'm OK with doing it in two parts. This PR brings the hook to the module and we test this in conjunction with the same change for ShortcutGuide, then we make the improvements.
Sounds like a plan?

Reply:
alright

Reply:
My idea was also to do it in two steps.